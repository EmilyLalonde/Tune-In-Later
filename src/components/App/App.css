body {
    background-image: url('https://free4kwallpapers.com/uploads/originals/2018/01/28/abstract-background-wallpaper.jpg');
    background-repeat: no-repeat;
    background-size: 100vw 150vh;
}

.welcome-h2 {
  color: #002236;
  font-size: 40px;
  text-align: center;
  margin-bottom: 2%;
  text-shadow: 1px 1px;
  text-shadow:
  0 0 10px #fff,
  0 0 15px #fff,
  0 0 30px #fff,
  0 0 50px #38bcf4,
  0 0 100px #38bcf4,
  0 0 130px #38bcf4,
  0 0 150px #38bcf4,
  0 0 170px #38bcf4;
}

.welcome-container {
  height: 90vh;
  overflow: scroll;
}

.sign-out-button {
  background-color: #36abdd;
  border: 3px solid #002236;
  border-radius: 6px;
  color: #002236;
  font-size: 20px;
  height: 50px;
  width: 100px;
}

.login-button:hover, 
.search-button:hover,
.create-user-button:hover,
.sign-out-button:hover {
  box-shadow:
   0 0 0.1vw  0.1vw #fff7f7,   
   0 0 0.3vw  0.3vw #002236, 
   0 0   1.5vw  0.2vw #36abdd,
   inset 0 0 1vw  0.2vw #36abdd,
   inset 0 0 0.3vw  0.1vw #002236,
   inset 0 0 0.1vw  0.1vw #fff7f7;
   background-color: rgba(0, 0, 0, .3)
}