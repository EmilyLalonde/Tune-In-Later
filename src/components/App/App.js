import React, { Component } from "react";
import { getAlbums, createUser, loginUser } from "../../apiCalls/apiCalls.js";
import { Route, NavLink, Redirect } from 'react-router-dom';
import FavoritesContainer from "../FavoritesContainer/FavoritesContainer";
import WelcomeContainer from "../WelcomeContainer/WelcomeContainer";
import LoginForm from '../LoginForm/LoginForm';
import CreateUserForm from '../CreateUserForm/CreateUserForm'
import Nav from '../Nav/Nav'
import "./App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      country: [],
      rock: [],
      pop: [],
      error: "",
      currentUser: null,
      redirect: true
    };
  }
  async componentDidMount() {
    try {
      const country = await getAlbums(1138);
      this.setState({ country });
    } catch ({ message }) {
      this.setState({ error: message });
    }

    try {const pop = await getAlbums(1151);
    this.setState({ pop });
    } catch ({ message }) {
      this.setState({ error: message})
    }

    try {const rock = await getAlbums(1071);
      this.setState({ rock });
      } catch ({ message }) {
        this.setState({ error: message})
      }
  }

  createTheUser = (user) => {
    createUser(user)
    .then(data => this.setState({currentUser: data}))
    .catch(err => this.setState({error: err.message}))
  }

  loginTheUser = (user) => {
    loginUser(user)
    .then(data => this.setState({currentUser: data}))
    .catch(err => this.setState({error: err.message}))
  }

  logoutUser =() => {
    this.setState({currentUser: null});
    console.log(this.state.currentUser)
  }

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <Nav currentUser={this.state.currentUser} handleLogout={this.logoutUser}/>
        <Route exact path='/' render={() => 
          <div>
            <h2>Doo-Wop</h2>
            <WelcomeContainer albums={this.state.country} />
            <h2>Hair Metal</h2>
            <WelcomeContainer albums={this.state.pop} />
            <h2>Gangsta Rap</h2>
            <WelcomeContainer albums={this.state.rock} />
          </div>
        }/>
        <Route exact path='/login' render={() => <LoginForm loginTheUser={this.loginTheUser}/>} />
        <Route exact path='/create-user' render={() => <CreateUserForm createTheUser={this.createTheUser}/>} />
      </div>
    );
  }
}

export default App;
