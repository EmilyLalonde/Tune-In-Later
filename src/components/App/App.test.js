import React from 'react';
import { shallow } from 'enzyme'
import { App } from './App';

describe('App', () => {

  let wrapper;
  beforeEach(() => {;
    const mockGetAlbums = jest.fn();
    const mockCreateUser = jest.fn();
    const mockLoginUser = jest.fn();
    const mockAddToFavorites = jest.fn();
    const mockGetFavorites = jest.fn();
    const mockDeleteFavorite = jest.fn();

    wrapper = shallow(<App 
      getAlbums={mockGetAlbums}
      createUser={mockCreateUser}
      loginUser={mockLoginUser}
      addToFavorites={mockAddToFavorites}
      getFavorites={mockGetFavorites}
      deleteFavorite={mockDeleteFavorite}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
})
