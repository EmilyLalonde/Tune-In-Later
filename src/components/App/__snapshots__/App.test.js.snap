// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <Nav
    handleLogout={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/create-user"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
</div>
`;
